packages/blade/src/components/Carousel/_decisions/decisions.md

```diff
@@ -0,0 +1,176 @@
+  <img src="./gst-page-carousel.png" width="50%" />
+
+
+> NOTE: To ensure bleed happens you need to specify a fixed amount of carousel item's width via the `carouselItemWidth` prop. Otherwise the carousel item will take up the 100% of the container width and won't bleed
+
+
+v1: visibleItems: autofit + auto bleed + shouldAddStartEndMargin
+
+<img src="./example-visibleitems-autofit-with-margin.png" width="70%" />
+
+v1: visibleItems: autofit + auto bleed
+> Note: in this variant, we won't show indicator because indicators won't be able to properly map to the visible slides since there can be any number of slides visible at a time based on the screenshide
+>
+<img src="./example-visibleitems-autofit-without-margin.png" width="70%" />
+
+### **Variant 2: With visibleItems:**
+
+This variant is a basic variant which can have: 
+
+- visibleItems: 1 | 2 | 3
+- The carouselItems inside doesn't have a fixed amount of width means they are fluid by default
+- Cannot bleed
+- Cannot have scroll overlay
+- Cannot have start/end margin
+
+
+v2: visibleItems: 1 
+
+<img src="./example-visibleitems-1.png" width="70%" />
+
+v2: visibleItems: 2 
+
+<img src="./example-visibleitems-2.png" width="70%" />
+
+v2: visibleItems: 3
+
+<img src="./example-visibleitems-3.png" width="70%" />
+
+Mobile variants: 
+
+On mobile screens if `visibleItems` is set to `1, 2 or 3` we automatically change it to `1`
+
+<img src="./example-mobile-visibleitems-1.png" width="70%" />
+
+And if `visibleItems` is set to `autofit` we don't do anything, and by changing the `carouselItemWidth` consumers can even get bleed on mobile: 
+
+```jsx
+// this will give 10% bleed on mobile
+<Carousel carouselItemWidth={{ base: '90%', m: '300px' }} />
+```
+
+<img src="./example-mobile-visibleitems-autofit.png" width="70%" />
+
+## Accessibility
+
+We need to follow the APG Pattern For [Carousel](https://www.w3.org/WAI/ARIA/apg/patterns/carousel/examples/carousel-2-tablist/)
+
+## Motion
+
+https://github.com/razorpay/blade/assets/35374649/6876f348-1b0a-4e9a-800a-300025f54b5c
+
+## Open Questions
+
+- Should we call it `navigationButtonStyle` or `navigationButtonVariant`? 
+- Should we make visibleItems: 'auto' or 'undefined'?
+  - We decided to use `autofit`
+
+## Libraries
+
+**web**
+
+On mweb & dweb we can use [CSS scroll snap](https://ishadeed.com/article/css-scroll-snap/) feature, this covers all our common usecase with bleed smooth scrolling etc.
+
+**react-native**
+
+on rnweb, we cannot use CSS scroll snap so might might need to opt for a library.
+But further exploration lead me to this [SO question](https://stackoverflow.com/questions/39849648/horizontal-scrollview-snapping-react-native) and [this expo example](https://snack.expo.io/H1CnjIeDb), turns our that ScrollView does support scroll snapping so if this works for our usecase it will be helpful.
+
+But we might need to see if this might cause an performance issue since we won't be using FlatList.
+
```

Comment:
How does the swipe behaviour work on mobile web ? Also if there are no indicators on dweb can clicking on the items will slide the items ?

Reply:
Yes swipe will be available. 

> Also if there are no navigation buttons on dweb can clicking on the items will slide the items ?
Nav buttons are mandatory in dweb.